### PR TITLE
Fixes #223 Use of Hooks::register was deprecated in MediaWiki 1.35

### LINKED ISSUE #223: PHP Deprecated:  Use of Hooks::register was deprecated in MediaWiki 1.35, PHP Deprecated:  Accessing $wgHooks directly is deprecated
### Setup and configuration

- SMW version: 4.1.2
- SESP version: 3.0.4
- MW version: 1.40.0
- PHP version: 8.1.2
- DB (MySQL etc. and version): 10.6.12-MariaDB-0ubuntu0.22.04.1

### Issue
```
PHP Deprecated:  Use of Hooks::register was deprecated in MediaWiki 1.35. [Called from SESP\HookRegistry::register in /var/www/wiki/extensions/SemanticExtraSpecialProperties/src/HookRegistry.php at line 36] in /var/www/wiki/includes/debug/MWDebug.php on line 382
```
```
PHP Deprecated:  Accessing $wgHooks directly is deprecated, use HookContainer::getHandlers() or HookContainer::register() instead. [Called from SemanticResultFormats::onExtensionFunction in /var/www/wiki/extensions/SemanticResultFormats/SemanticResultFormats.php at line 97] in /var/www/wiki/includes/debug/MWDebug.php on line 382
```

Steps to reproduce the observation (recommendation is to use the [sandbox](https://sandbox.semantic-mediawiki.org)):

php maintenance/runJobs.php

### PR DESCRIPTION
**Pull request by JLTRY**
This PR is made in reference to: #223

### CODE REVIEW

File: src/HookRegistry.php [hunk 1]
```diff
@@ -32,8 +32,9 @@ public function __construct( $config ) {
 	 * @since  1.0
 	 */
 	public function register() {
+		$hookContainer = MediaWikiServices::getInstance()->getHookContainer();
 		foreach ( $this->handlers as $name => $callback ) {
-			Hooks::register( $name, $callback );
+			$hookContainer->register( $name, [ $this, $callback ]);
```
Comment (JeroenDeDauw) [inline src/HookRegistry.php:37]: Why not

```php
$hookContainer->register( $name, $callback );
```

Comment (JLTRY) [reply src/HookRegistry.php]: I pushed an other commit

Comment (JeroenDeDauw) [reply src/HookRegistry.php]: Did you test this code?

Comment (JLTRY) [reply src/HookRegistry.php]: hello yes locally please take both 2 commits

**Pull request merged**